shiny/ui/_theme_brand.py

```diff
@@ -318,45 +236,72 @@ def _get_theme_name(self, brand: "Brand") -> str:
     @staticmethod
     def _prepare_color_vars(
         brand: "Brand",
-    ) -> tuple[dict[str, str], dict[str, str], list[str]]:
-        """Colors: create a dictionary of Sass variables and a list of brand CSS variables"""
+    ) -> tuple[dict[str, YamlScalarType], dict[str, YamlScalarType], list[str]]:
+        """
+        Colors: Create a dictionaries of Sass and CSS variables
+        """
         if not brand.color:
             return {}, {}, []
 
-        mapped: dict[str, str] = {}
-        brand_sass_vars: dict[str, str] = {}
-        brand_css_vars: list[str] = []
+        defaults_dict: dict[str, YamlScalarType] = {}
+        palette_defaults_dict: dict[str, YamlScalarType] = {}
+        palette_css_vars: list[str] = []
 
-        # Map values in colors to their Sass variable counterparts
         for thm_name, thm_color in brand.color.to_dict(include="theme").items():
-            if thm_name not in color_map:
-                warn_or_raise_unmapped_variable(f"color.{thm_name}")
-                continue
-
-            for sass_var in color_map[thm_name]:
-                mapped[sass_var] = thm_color
+            # Create brand Sass variables and set related Bootstrap Sass vars
+            # brand.color.primary = "#007bff"
+            # ==> $brand_color_primary: #007bff !default;
+            # ==> $primary: $brand_color_primary !default;
+
+            brand_color_var = f"brand_color_{thm_name}"
+            defaults_dict[brand_color_var] = thm_color
+            # Currently, brand.color fields are directly named after Bootstrap vars. If
+            # that changes, we'd need to use a map here. These values can't be set to
+            # `null !default` because they're used by maps in the Bootstrap mixins layer
+            # and cause errors if a color is `null` rather than non-existent.
```

Comment:
Could it be that this is the problem? https://github.com/rstudio/bslib/blob/112efd0e/inst/lib/bs5/scss/_variables.scss#L303

Reply:
Yup! Fixed in rstudio/bslib#1149